File: app/src/main/java/com/evaluation/dagger/data/NavigatorModule.java
package com.evaluation.dagger.data;

import com.evaluation.navigation.Navigator;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author Vladyslav Havrylenko
 * @since 09.03.2020
 */
@Module
class NavigatorModule {

    private static NavigatorModule instance;

    NavigatorModule() {
    }

    static NavigatorModule getInstance() {
        if (instance == null) {
            instance = new NavigatorModule();
        }
        return instance;
    }

    @Provides
    @Singleton
    Navigator provideNavigator() {
        return new Navigator();
    }
}

File: app/src/main/java/com/evaluation/dagger/data/NavigatorModule.kt
package com.evaluation.dagger.data

import com.evaluation.navigation.Navigator
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author Vladyslav Havrylenko
 * @since 10.03.2020
 */
@Module
object  NavigatorModule {
    @Provides
    @Singleton
    fun provideNavigator(): Navigator = Navigator()
}